Bluntsord/added new instructions

Bluntsord opened the pull request:
Added some "Scoping" which is basically some spotlight like thing

Bluntsord committed:
Added some more instructions after prompt

Bluntsord committed:
Added the queue free for the scoping

Bluntsord committed:
added back button

Bluntsord committed:
Changed some english

bharathcs commented on game/src/Scope/TowerButtonScopes.gd:
```diff
@@ -0,0 +1,9 @@
+var instructions = "These are the tower buttons\n. " \
+	+ "Click here\n to choose the tower built\n " \
+	+ "Currently only the first tower is available\n" \
+	+ "Alternatively you can press Q on your keyboard"
```
Hmm Q swaps colour tho?

bharathcs replied:
tc fixed.

bharathcs commented on game/project.godot:
```diff
@@ -93,6 +93,11 @@ _global_script_classes=[ {
 "path": "res://src/ui/gameplay/towerInventory/TowerInventorySlot.gd"
+}, {
+"base": "AnimatedSprite",
+"class": "prompt",
```
Oof could classes start with capital letter please

bharathcs replied:
Fixed in f802cd13

web-flow committed:
Let the code elf take a look.

bharathcs committed:
Make minor comprehension edits.

Rename the AnimatedSprite.gd to Prompts.gd and fix class name to Prompts instead of prompts

Change animation names to reflect the use.

Rename 'Temp' in menu to 'Logo'

bharathcs commented on game/src/Scope/GoldScoping.gd:
```diff
@@ -0,0 +1,10 @@
+extends Label
+
+var instructions = "This is gold you have\n " \
```
new:

This is the gold you have:
Killing enemies drop gold and
Building towers cost gold

That's all for now. Have fun!

Missing spaces, punctuation, capitalisation, and grammar fix.

bharathcs commented on game/src/Scope/HealthBarScopeLabel.gd:
```diff
@@ -0,0 +1,9 @@
+var instructions = "This is the health bar" \
+	+ "The red bar indicates the health\n of the red portal " \
+	+ "And the blue health\n indicates the health of the blue portal\n" \
+	+ "If any reaches zero, GAMEOVER"
```
GAMEOVER -> GAME OVER

bharathcs replied:
tc fixed.

Bluntsord committed:
Merge pull request #125 from quantum-defence/gdformatter-1627138932

Format GD script

bharathcs commented on game/src/Scope/ItemScoping.gd:
```diff
@@ -0,0 +1,10 @@
+var instructions = "These are the items. Items dropped by enemies\n " \
+	+ "can be clicked on and will appear here\n" \
+	+ "Open a tower inventory by clicking on a tower\n" \
+	+ "and equip an item by dragging and dropping"
```
```
var instructions = "These are the items. Items dropped by enemies\n " \
	+ "can be clicked on and will appear here\n" \
	+ "Open a tower inventory by clicking on a tower\n" \
// better?:
var instructions = "Click on the items dropped by enemies\n " \
	+ "to have them appear here." .....
```

bharathcs replied:
tc fixed

bharathcs commented on game/src/Scope/QuantumButton.gd:
```diff
@@ -0,0 +1,10 @@
+extends Label
+
+var instructions = "This is the quantum button. " \
+	+ "Click here\n to swap between color" \
```
er minor thing but it should probably be colour since the name of the game is in British English (the c in defence)

bharathcs replied:
tc fixed.

bharathcs commented on game/src/environment/Fog/Black.shader:
```diff
@@ -0,0 +1,7 @@
+shader_type canvas_item;
+
+uniform vec4 color = vec4(0, 0, 0, 1);
```
I do believe this variable should be renamed as black

bharathcs replied:
Never mind I realised you did color everywhere

bharathcs replied:
tc ignored.

bharathcs commented on game/src/ui/gameplay/Prompts/AnimatedSprite.gd:
```diff
@@ -1,30 +1,51 @@
+	"This lengthens the time\n you can attack them",
+	"Build towers to defend your portal\n the gold of each \ntower is displayed at the top",
 	"However BEWARE Quantum towers\n do not do any damage!, they only\n change the color of enemies"
 ]
```
```
"Protect your portals!\n Red enemies go to red portal\nand blue enemies go\nto blue portal. Each tower\ncan only hit enemies \nof the same color",
	"Add quantum items to the towers \nto allow enemies to change \nbetween colors",
	"Hadamard(Orange) gives a \n50/50 chance",
	"RY item(Green) gives a \n12.5/87.5 chance",
	"And the classical X gate(Blue)\n is a 0/100 chance",
	"Enemies that change color\n are forced to walk to\n the new colors portal",
	"This lengthens the time\n u can attack them",
	"Build towers to defend ur portal\n the gold of each \ntower is displayed at the top",
	"This lengthens the time\n you can attack them",
	"Build towers to defend your portal\n the gold of each \ntower is displayed at the top",
	"However BEWARE Quantum towers\n do not do any damage!, they only\n change the color of enemies"
```

Protect your portals!
Red enemies attack through the red portal
and blue enemies attack through the blue portal. Enemy colour is set by the qubit they contain, and each tower only hits enemies of the same colour. 

Add quantum items to the towers to ~~allow enemies to change~~ force an enemy into a colour based on the quantum circuitry in the item. If it flips them into the opposite colour, the enemy is forced to turn around and go to its new target, which lets you rack up more hits along its path. However, BEWARE that these quantum towers do not do any damage, and they only affect the qubit (colour) of the enemy.

Hadamard (Orange) will put the enemy back into equal superposition before the weapon collapses it. A **red** tower equipped with Hadamard will turn on average 50% of the red enemies it hits to blue.

RY item (Green) puts an enemy into a state of superposition that skews towards collapsing into red: 87.5% chance of switching to red to be precise.

The classical X gate (Blue) will simply flip the qubit 100% of the time, turning all enemies it hits into the opposite colour.

Build towers to defend your portal! The cost of each tower is displayed on the right.

bharathcs replied:
grammar, rephrased cuz you're not allowing enemies anything, you and your vs u and ur, rewriting items to be easier to understand, spaces and punctuation

Bluntsord committed:
Changed instructions and fixed some minor bug

Bluntsord committed:
Changed greyout abit

bharathcs commented on game/src/Scope/GoldScoping.gd:
```diff
@@ -1,10 +1,10 @@
+	"This is the gold you have:"
+	+ "Killing enemies drop gold and"
+	+ "Build towers cost gold."
+	+ "Thats all for now. Have Fun!"
```
fun

bharathcs commented on game/src/Scope/QuantumButton.gd:
```diff
@@ -2,9 +2,9 @@ extends Label
-	+ "Click here\n to swap between color"
-	+ "of towers built.\n Remember that red towers"
-	+ "can only \n hit red enemies and vice versa\n"
+	+ "Click here to swap between colour"
```
colours

bharathcs reviewed (changes requested):
there are still so many simple problems with spaces, punctuation, grammar, and i also see an empty file GoldCost.gd, print statements and other things

bharathcs committed:
Fix things tc couldn't be bothered with.

print statements, capitalisations, spaces, punctuation, singular / plural, empty files, etc.
